cdap-gateway/src/main/java/co/cask/cdap/gateway/router/NettyRouter.java

```diff
@@ -154,22 +140,20 @@ public NettyRouter(CConfiguration cConf, SConfiguration sConf, @Named(Constants.
                                                      sConf.get(Constants.Security.Router.SSL_KEYSTORE_PASSWORD),
                                                      sConf.get(Constants.Security.Router.SSL_KEYPASSWORD));
     } else {
-      this.serviceToPortMap.put(Constants.Router.GATEWAY_DISCOVERY_NAME,
-                        Integer.parseInt(cConf.get(Constants.Router.ROUTER_PORT,
-                                                   Constants.Router.DEFAULT_ROUTER_PORT)));
+      this.serviceToPortMap.put(Constants.Router.GATEWAY_DISCOVERY_NAME, cConf.getInt(Constants.Router.ROUTER_PORT));
       if (webAppEnabled) {
-        this.serviceToPortMap.put(Constants.Router.WEBAPP_DISCOVERY_NAME,
-                                  Integer.parseInt(cConf.get(Constants.Router.WEBAPP_PORT,
-                                                             Constants.Router.DEFAULT_WEBAPP_PORT)));
+        this.serviceToPortMap.put(Constants.Router.WEBAPP_DISCOVERY_NAME, cConf.getInt(Constants.Router.WEBAPP_PORT));
       }
       this.sslHandlerFactory = null;
     }
+    this.connectionTimeout = cConf.getInt(Constants.Router.CONNECTION_TIMEOUT_SECS);
+    this.timer = new HashedWheelTimer();
```

Comment:
Also, good to provide a `ThreadFactory` so that the thread created is named properly.